src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs

```diff
@@ -303,6 +303,19 @@ IFormManager formManager
                 });
             }
 
+            foreach (var workflowRecord in workflowDefinitionRecord.WorkflowRecords) {

```

Comment:
@j3ffb just to say that now, just before, the workflowDefinitionRecord.WorkflowRecords collection is already cleared. This since this commit: https://github.com/OrchardCMS/Orchard/commit/1b8e00a9c60cddc0d1e69d867abb44bac1526e9f

Now, when you edit and save a workflow, if they are some running instances, a confirm dialog is displayed. If confirmed, all running instances are deleted, and their related awaiting activities...

Best